Remotion/Web/Core/UI/Controls/WebTreeNodeCollection.cs

```diff
@@ -108,5 +110,22 @@ public new WebTreeNode Find (string id)
     //        node.CollapseAll();
     //    }
     //  }
+
+    protected internal IEnumerable<WebTreeNode> GroupByCategory ()
+    {
+      return GroupByCategory (InnerList.Cast<WebTreeNode>().ToArray());
+    }
+
+    private static IEnumerable<WebTreeNode> GroupByCategory (WebTreeNode[] nodes)
+    {
+      var categories = nodes.Select (node => node.Category).Distinct();
+      var orderedNodes = new List<WebTreeNode>();
+      foreach (var category in categories)
+      {
+        orderedNodes.AddRange (nodes.Where (node => node.Category == category));
+      }
+
+      return orderedNodes;
+    }
```

Comment:
```suggestion
    public static IReadOnlyCollection<WebTreeNode> GroupByCategory (WebTreeNode[] nodes)
    {
      var categories = nodes.Select (node => node.Category).Distinct();
      var orderedNodes = new List<WebTreeNode>();
      foreach (var category in categories)
      {
        orderedNodes.AddRange (nodes.Where (node => node.Category == category));
      }

      return orderedNodes;
    }
```

Also, statics on top, add doc comments and argument checks

Reply:
According to the docs (https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.groupby?view=netframework-4.8#System_Linq_Enumerable_GroupBy__2_System_Collections_Generic_IEnumerable___0__System_Func___0___1__System_Collections_Generic_IEqualityComparer___1__) Linq GroupBy also does a stable grouping. `GroupBy(...).SelectMany(...).ToArray()` is probably the cleanest approach.